src/record.rs

```diff
@@ -79,41 +79,41 @@ impl MSSampleType {
 }
 
 /// An enumeration of possible data encodings.
-#[repr(i8)]
+#[repr(u8)]
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 pub enum MSDataEncoding {
     /// Text encoding (UTF-8)
-    Text = raw::DE_TEXT as c_char,
+    Text = raw::DE_TEXT as u8,
     /// 16-bit integer encoding
-    Integer16 = raw::DE_INT16 as c_char,
+    Integer16 = raw::DE_INT16 as u8,
     /// 32-bit integer encoding
-    Integer32 = raw::DE_INT32 as c_char,
+    Integer32 = raw::DE_INT32 as u8,
     /// 32-bit floating point encoding (IEEE)
-    Float32 = raw::DE_FLOAT32 as c_char,
+    Float32 = raw::DE_FLOAT32 as u8,
     /// 64-bit floating point encoding (IEEE)
-    Float64 = raw::DE_FLOAT64 as c_char,
+    Float64 = raw::DE_FLOAT64 as u8,
     /// Steim-1 compressed integer encoding
-    Steim1 = raw::DE_STEIM1 as c_char,
+    Steim1 = raw::DE_STEIM1 as u8,
     /// Steim-2 compressed integer encoding
-    Steim2 = raw::DE_STEIM2 as c_char,
+    Steim2 = raw::DE_STEIM2 as u8,
     /// **Legacy**: GEOSCOPE 24-bit integer encoding
-    GeoScope24 = raw::DE_GEOSCOPE24 as c_char,
+    GeoScope24 = raw::DE_GEOSCOPE24 as u8,
     /// **Legacy**: GEOSCOPE 16-bit gain ranged, 3-bit exponent
-    GeoScope163 = raw::DE_GEOSCOPE163 as c_char,
+    GeoScope163 = raw::DE_GEOSCOPE163 as u8,
     /// **Legacy**: GEOSCOPE 16-bit gain ranged, 4-bit exponent
-    GeoScope164 = raw::DE_GEOSCOPE164 as c_char,
+    GeoScope164 = raw::DE_GEOSCOPE164 as u8,
     /// **Legacy**: CDSN 16-bit gain ranged
-    CDSN = raw::DE_CDSN as c_char,
+    CDSN = raw::DE_CDSN as u8,
     /// **Legacy**: SRO 16-bit gain ranged
-    SRO = raw::DE_SRO as c_char,
+    SRO = raw::DE_SRO as u8,
     /// **Legacy**: DWWSSN 16-bit gain ranged
-    DWWSSN = raw::DE_DWWSSN as c_char,
+    DWWSSN = raw::DE_DWWSSN as u8,
 }
 
 impl MSDataEncoding {
     /// Create a `MSDataEncoding` from the given `ch`.
-    pub fn from_char(ch: c_char) -> MSResult<Self> {
-        match ch as c_uint {
+    pub fn from_char(ch: u8) -> MSResult<Self> {
+        match ch as u32 {
```

Comment:
Similar to above question, is this cast needed?

Reply:
yes, we need this cast or else it generates complier error

Reply:
OK, thanks, I believed that the values that follow were unsuffixed literals, but from your comment above, "bindgen" has converted them to `u32`